eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/io/DbsnpVariantsWriter.java

```diff
@@ -179,9 +182,107 @@ public void write(List<? extends DbsnpVariantsWrapper> wrappers) throws Exceptio
                        .collect(Collectors.toList());
     }
 
-    private void writeClusteredVariantsDeclustered(List<DbsnpClusteredVariantEntity> clusteredVariantsDeclustered) {
-        if (!clusteredVariantsDeclustered.isEmpty()) {
-            dbsnpClusteredVariantDeclusteredWriter.write(clusteredVariantsDeclustered);
+    private List<DbsnpSubmittedVariantOperationEntity> updateClusteredVariantAccessionsInSubmittedVariants(
+            List<? extends DbsnpVariantsWrapper> wrappers,
+            List<DbsnpClusteredVariantOperationEntity> mergeClusteredOperations) {
+
+        Map<String, Long> replacements = getAccessionReplacements(mergeClusteredOperations);
+        List<DbsnpSubmittedVariantOperationEntity> operations = new ArrayList<>();
+
+        for (DbsnpVariantsWrapper wrapper : wrappers) {
+            updateClusteredVariantAccessionsInSubmittedVariants(wrapper, replacements, operations);
+        }
+        return operations;
+    }
+
+    /**
+     * @return a mapping of where a clustered variant was merged: map from hash+accession to which accession it was
+     * merged into.
+     */
+    private Map<String, Long> getAccessionReplacements(
+            List<DbsnpClusteredVariantOperationEntity> mergeClusteredOperations) {
+        Map<String, List<DbsnpClusteredVariantOperationEntity>> accessionToOperationsMap = new HashMap<>();
+        for (DbsnpClusteredVariantOperationEntity mergeClusteredOperation : mergeClusteredOperations) {
+            String hash = mergeClusteredOperation.getInactiveObjects().get(0).getHashedMessage();
+            Long accession = mergeClusteredOperation.getAccession();
+            accessionToOperationsMap
+                    .computeIfAbsent(buildKeyFromHashAndAccession(hash, accession), k -> new ArrayList<>())
+                    .add(mergeClusteredOperation);
+        }
+
+        Map<String, Long> oldToNewAccessions = new HashMap<>();
+        accessionToOperationsMap.forEach((accession, accessionOperations) -> {
+            List<Long> mergedIntoList = accessionOperations
+                    .stream()
+                    .map(EventDocument::getMergedInto)
+                    .distinct()
+                    .collect(Collectors.toList());
+
+            if (mergedIntoList.size() > 1) {
+                throw new IllegalStateException(
+                        "Clustered Variant rs" + accession + " was merged into several other clustered Variants with "
```

Comment:
it's not so simple, this map is built with the list of operations (not the `ClusteredVariantEntity`), but later we query it having the `ClusteredVariantEntity`, so we can not put any of those classes as key.

this could be changed if we did the writing of the clustered variants one by one, instead of all of them in a chunk.

I will use a [Pair](https://docs.spring.io/spring-data/commons/docs/current/api/org/springframework/data/util/Pair.html) as key.